element-hq/element-x-android
# Media gallery UI

## Content


Implement first version of media gallery.

Figma: https://www.figma.com/design/Ni6Ii8YKtmXCKYNE90cC67/Timeline-(new)?m=dev 

This PR add new screens to be able to see all the media of a room: images and videos as a gallery (grid) and files as a list.

Will be done later:
- render audio item in another way in the file list (see Figma)
- month separators instead of day separator (waiting for change in the SDK api)
- swipe media to left and right in the media viewer, to navigate through the images / videos.
- search for media

## Motivation and context



## Screenshots / GIFs



## Tests



- From the timeline, click on an image Event.
- Click on "info" picto and see the new bottom sheet

- From the room detail screen, click on "Media and files"
- See the gallery of images, that can be scrolled to load more media. It should load automatically until the page is full or the beginning of the room is reached
- long click on an image to see the bottom sheet
- click on an image to see it fullscreen
- navigate to the file list
- test the action on file
- click on file

## Tested devices

- OS version(s):

## Checklist
